### DISCUSSION TITLE
More prominent custom fields within issue views

### DISCUSSION THREAD

**Discussion by chrisui** [community/community#6090] [Projects and Issues]
Really great improvement having custom fields on project cards which are accessible through issues but would love for them to be more prominent. Given it seems many people have a one-to-many issue-to-project ratio perhaps it needs to be configured per repository, user, or organisation. Even just remembering whether I (a user) has "expanded" the custom field view on issues would be great.

**Comment by RickCogley** [comment]
At least, being able to pin open the task custom field area would be helpful, but the best thing would be if those fields could be considered "first class citizens" in an issue.

<img width="436" alt="image" src="https://user-images.githubusercontent.com/512328/137604990-23f4afa6-c096-43aa-8507-b26249a45c12.png">

**Comment by clausreinke** [comment]
As you mention, the "one-to-many issue-to-project ratio" complicates things a little, as each project overlays a user's (or organisation's) view  on the issues. It might be worth expanding on that:

- that makes custom fields as "first class citizens" a separate issue - it would be helpful if the issue/repo owner could add project-independent custom fields (https://github.com/github/feedback/discussions/5755), in addition to the project-dependent custom fields added by project owners
- there might need to be a "subscribe to project" option, where issue viewers can decide which projects they want to see in an issue view
  - if the issue/repo owner adds an issue to a project, I probably want to see the project's custom fields as a default
  - if I add an issue to one of my projects, I definitely want to see my custom fields
  - if someone else adds an issue to one of their projects, I might want to see its custom fields, or not
- for my projects, I want control over which custom fields should be open in issues, at least as a default setting
- for my subscribed projects, I want control over which custom fields should be open in issues, overriding defaults
